src/database/alert_dao.rs

```diff
@@ -0,0 +1,536 @@
+use super::database::Database;
+use crate::{
+    errors::ServicesError,
+    expect_log,
+    reporting::{
+        public_report::{CoughSeverity, FeverSeverity, PublicReport},
+        symptom_inputs::UserInput,
+    },
+    reports_interval,
+    reports_update::reports_updater::Alert,
+};
+use log::*;
+use reports_interval::UnixTime;
+use rusqlite::{params, Row, NO_PARAMS};
+use std::sync::Arc;
+
+pub trait AlertDao {
+    fn all(&self) -> Result<Vec<Alert>, ServicesError>;
+    fn save(&self, alerts: Vec<Alert>) -> Result<(), ServicesError>;
+    fn delete(&self, id: String) -> Result<(), ServicesError>;
+}
+
+pub struct AlertDaoImpl {
+    db: Arc<Database>,
+}
+
+impl AlertDaoImpl {
+    pub fn new(db: Arc<Database>) -> AlertDaoImpl {
+        Self::create_table_if_not_exists(&db);
+        AlertDaoImpl { db }
+    }
+
+    fn create_table_if_not_exists(db: &Arc<Database>) {
+        // TODO use blob for tcn? https://docs.rs/rusqlite/0.23.1/rusqlite/blob/index.html
```

Comment:
SQLite is not strongly typed: it uses [dynamic typing/type affinity](https://www.sqlite.org/datatype3.html). In addition BLOB is meant to be used for large objects and our tcns do not fit that description.
TEXT type should be fine for tcns.

Reply:
I initially also thought that it wasn't suitable ("binary large object"), but I remember not finding a good reason. I also asked in a SQLite chat and someone that seemed experienced said that it was suitable. The point is to avoid serialization to hex. We should compare performance.

Type affinity seems mostly not relevant. This is about the format we should use for the data.